nltk/stem/paicehusk.py

```diff
@@ -0,0 +1,274 @@
+"""
+Paice-Husk Stemmer
+
+If you just want to use the standard Paice-Husk stemming rules, use the
+module's stem()` function::
+
+    stemmed_word = stem(word)
+
+If you want to use a custom rule set, read the rules into a string where the
+rules are separated by newlines, and instantiate the object with the string,
+then use the object's stem method to stem words::
+
+    stemmer = PaiceHuskStemmer(my_rules_string)
+    stemmed_word = stemmer.stem(word)
+
+"""
+
+import re
+from collections import defaultdict
+
+
+class PaiceHuskStemmer(object):
+    """
+    PaiceHuskStemmer
+
+        >>> from nltk.stem.paicehusk import PaiceHuskStemmer
+        >>> st = PaiceHuskStemmer()
+        >>> st.stem("determined")
+        'determin'
+        >>> st.stem('maximum')     # Remove "-um" when word is intact
+        'maxim'
+        >>> st.stem('presumably')  # Don't remove "-um" when word is not intact
+        'presum'
+        >>> st.stem('multiply')    # No action taken if word ends with "-ply"
+        'multiply'
+        >>> st.stem('provision')   # Replace "-sion" with "-j" to trigger "j" set of rules
+        'provid'
+        >>> st.stem('owed')        # Word starting with vowel must contain at least 2 letters
+        'ow'
+        >>> st.stem('ear')         # ditto
+        'ear'
+        >>> st.stem('saying')      # Words starting with consonant must contain at least 3
+        'say'
+        >>> st.stem('crying')      #     letters and one of those letters must be a vowel
+        'cry'
+        >>> st.stem('string')      # ditto
+        'string'
+        >>> st.stem('meant')       # ditto
+        'meant'
+        >>> st.stem('cement')      # ditto
+        'cem'
+    """
```

Comment:
Cool! When we have more of them maybe we can add a wiki on different effects of different stemmers =)